src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs

```diff
@@ -86,7 +86,21 @@ public ResultWrapper<ReceiptForRpc[]> parity_getBlockReceipts(BlockParameter blo
             TxReceipt[] receipts = _receiptFinder.Get(block) ?? new TxReceipt[block.Transactions.Length];
             bool isEip1559Enabled = _specProvider.GetSpec(block.Number).IsEip1559Enabled;
             IEnumerable<ReceiptForRpc> result = receipts.Zip(block.Transactions, (r, t) => new ReceiptForRpc(t.Hash, r, t.CalculateEffectiveGasPrice(isEip1559Enabled, block.BaseFeePerGas)));
-            return ResultWrapper<ReceiptForRpc[]>.Success(result.ToArray());
+            ReceiptForRpc[] resultAsArray = result.ToArray();
+            MakeLogIndexesUniqueInBlock(resultAsArray);
+            return ResultWrapper<ReceiptForRpc[]>.Success(resultAsArray);
+        }
+
+        private static void MakeLogIndexesUniqueInBlock(ReceiptForRpc[] resultAsArray)
+        {
+            int logIndexInBlock = 0;
+            for (int receiptIndex = 0; receiptIndex < resultAsArray.Length; receiptIndex++)
+            {
+                for (int logIndexInTx = 0; logIndexInTx < resultAsArray[receiptIndex].Logs.Length; logIndexInTx++)
+                {
+                    resultAsArray[receiptIndex].Logs[logIndexInTx].LogIndex = logIndexInBlock++;
+                }
+            }
         }
```

Comment:
We need to check if we don't need this in other RPC methods too

Reply:
@kjazgar will recheck the rest of endpoints, but for now it looks good for me.

Reply:
@kjazgar @MarekM25 can you check eth_getTransactionReceipt?